tests/unit/components/sl-date-range-picker-test.js

```diff
@@ -0,0 +1,352 @@
+
+    Ember.run( () => {
+        this.$( '.sl-daterange-start-date input' ).trigger( 'change' );
+    });
+});
+
+test( 'Earliest end date is the based on min date and start date', function( assert ) {
+    let component = this.subject();
+
+    assert.strictEqual(
+        component.get( 'earliestEndDate' ),
+        null
+    );
+
+    component.set( 'minDate', '01/01/2001' );
+
+    assert.equal(
+        component.get( 'earliestEndDate' ),
+        '01/01/2001'
+    );
+
+    component.set( 'startDateValue', '01/01/2015' );
+
+    assert.equal(
+        component.get( 'earliestEndDate' ),
+        '01/01/2015'
+    );
+});
+
+test( 'Latest start date is the based on max date and end date', function( assert ) {
+    let component = this.subject();
+
+    assert.strictEqual(
+        component.get( 'latestStartDate' ),
+        null
+    );
+
+    component.set( 'maxDate', '01/01/2029' );
+
+    assert.equal(
+        component.get( 'latestStartDate' ),
+        '01/01/2029'
+    );
+
+    component.set( 'endDateValue', '01/01/2015' );
+
+    assert.equal(
+        component.get( 'latestStartDate' ),
+        '01/01/2015'
+    );
+});
+
+test( 'Events from start date input are removed upon willClearRender', function( assert ) {
+    let component = this.subject();
+    let startDateInput = this.$( '.sl-daterange-start-date input' )[0];
+
+    assert.equal(
+        Ember.typeOf( $._data( startDateInput, 'events' ).change ),
+        'array',
+        'Start date input has change event listener after render'
+    );
+
+    component.trigger( 'willClearRender' );
+
+    assert.strictEqual(
+        $._data( startDateInput, 'events' ),
+        undefined,
+        'Start date input has no event listeners after willClearRender'
+    );
+});
+
+test( 'label, startDatePlaceholder, and endDatePlaceholder are undefined by default', function( assert ) {
+    let component = this.subject();
+
+    assert.strictEqual(
+        component.get( 'label' ),
+        undefined,
+        'label is undefined by default'
+    );
+
+    assert.strictEqual(
+        this.$( 'label' )[0],
+        undefined,
+        'No label is created when label is undefined'
+    );
+
+    assert.strictEqual(
+        component.get( 'startDatePlaceholder' ),
+        undefined,
+        'startDatePlaceholder is undefined by default'
+    );
+
+    assert.strictEqual(
+        this.$( '.sl-daterange-start-date input' ).prop( 'placeholder' ),
+        '',
+        'Start date input placeholder is empty when startDatePlaceholder is undefined'
+    );
+
+    assert.strictEqual(
+        component.get( 'endDatePlaceholder' ),
+        undefined,
+        'endDatePlaceholder is undefined by default'
+    );
+
+    assert.strictEqual(
+        this.$( '.sl-daterange-end-date input' ).prop( 'placeholder' ),
+        '',
+        'End date input placeholder is empty when endDatePlaceholder is undefined'
+    );
+});
+
+test( 'label is accepted as a parameter', function( assert ) {

```

Comment:
There should also be a test that confirms that a `label` element is not rendered when the `label` property is not populated